Add fuzzy auto length support for NEST

SukharevAndrey opened the pull request:
As of now fuzziness AUTO parameter does not support custom length behaviour added in https://github.com/elastic/elasticsearch/issues/25528 and described on https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness

This PR contains simple implementation for that to NEST. It could be easily ported to the new JSON serialization implementation: https://github.com/elastic/elasticsearch-net/pull/3583

russcam commented on src/Nest/CommonOptions/Fuzziness/FuzzinessJsonConverter.cs:
```diff
@@ -1,19 +1,28 @@
 {
 	internal class FuzzinessJsonConverter : JsonConverter
 	{
+		private static readonly Regex AutoLengthRegex = new Regex(@"^AUTO:(?<low>\+?\d+),(?<high>\+?\d+)$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
I suspect looking for the index of `:` and `,` will be faster than a regex here.

SukharevAndrey replied:
Changed. it is now more simple, but may throw an exception when source string is malformed (for example when ```,``` is before ```;```

russcam replied:
I think this is acceptable here, since Elasticsearch constrains the format of the response:

https://github.com/elastic/elasticsearch/blob/3b8c2e4d8e221677324753cd3265cc111d0faa95/server/src/main/java/org/elasticsearch/common/unit/Fuzziness.java#L69-L77

SukharevAndrey commented:
Thank you for your feedback, @russcam.
Changed the implementation, and it is now more than 6 times faster :)
Also added docs for parameters based on official documentation and included usage of AutoLength in  MatchUsageTests.

russcam commented on src/Nest/CommonOptions/Fuzziness/Fuzziness.cs:
```diff
@@ -3,15 +3,41 @@
+		/// </summary>
+		/// <param name="low">Must match exactly for terms with less length</param>
+		/// <param name="high">Two edits allowed for terms with greater length</param>
+		public static Fuzziness AutoLength(int low, int high) => new Fuzziness
```
Documentation 💯

russcam commented:
ported to 6.x: 167fa29e34b50d853d11f709e517a1630b84662c
